### PR TITLE
Include dcat:downloadURL on release and revision

### LINKED ISSUE #344: Include dcat:downloadURL on release and revision
We should include direct links in the JSON/LD to the supported direct downloads.

We should follow dcat here e.g. check the predicates:

```json
{
  "contents": [
     // release
    {"@id":"2019",
       "dcat:distribution": {"text/csv": {"@id": "http://..../the-location.csv", "dcat:downloadUrl": "http://..../the-location.csv"}, "application/json": {"@id":  "http://..../the-location.json", "dcat:downloadUrl": "http://..../the-location.json"}, {"@id":"http://..../the-location/release-metadata.csv.json", "application/csvm+json": "http://..../the-location/release-metadata.csv.json"}

    }
  ],
}
```

We should ideally use [JSON/LD index maps](https://w3c.github.io/json-ld-syntax/#data-indexing) in our context to model the syntax here.

Update the `@context` for the `dcat:distribution` predicate to allow this type of syntax, whilst still yielding triples of the expected form, e.g.

```turtle
</2019> dcat:distribution <http://..../the-location.csv> .
<http://..../2019/the-location.csv> dcat:downloadUrl "http://..../2019/the-location.csv" .
```

### PR DESCRIPTION
**Pull request by andrewmcveigh**
Fix: #344

### CODE REVIEW

File: datahost-ld-openapi/test/tpximpact/datahost/ldapi/models/release_test.clj [hunk 1]
```diff
@@ -249,3 +257,49 @@
...
+              json (distribution "application/json")
+              csvm (distribution "application/csvm+json")]
+          (is (= 200 (:status response)))
+          (is (= (str path-id ".csv") (csv "@id")))
+          ;; Is this ^^ right? compacting seems to drop http://localhost:3400/data
+          (is (= (str "http://localhost:3400" release-1-path ".csv")
+                 (csv "dcat:accessURL")))
+
+          (is (= (str path-id ".json") (json "@id")))
+          ;; Is this ^^ right? compacting seems to drop http://localhost:3400/data
```
Comment (tpximpact-rickmoynihan) [inline datahost-ld-openapi/test/tpximpact/datahost/ldapi/models/release_test.clj:298]: Looks right to me, but we could add an additional assertion that concatenates the `@context @base` with `path-id` and check that is the expected full URI.

Comment (RickMoynihan) [inline datahost-ld-openapi/test/tpximpact/datahost/ldapi/models/release_test.clj]: Looks right to me, but we could add an additional assertion that concatenates the @context @base with path-id and check that is the expected full URI.

**Comment by kjoshi** [comment]
I guess if #380 gets merged then the `:dcat:distribution` parts added to the context in `json_ld.clj` can be moved into [jsonld-context.json](https://github.com/Swirrl/datahost-prototypes/blob/3460946ee7325b4495a82e6a7f7f11ba21e560b4/datahost-ld-openapi/resources/jsonld-context.json)?

**Comment by xdrcft8000** [comment]
JSON response for revision: 

<pre>
{
  "dh:hasChange": [
    "https://example.org/data/English-Indices-of-Deprivation/release/2019/revision/1/commit/1"
  ],
  "dh:appliesToRelease": "https://example.org/data/English-Indices-of-Deprivation/release/2019",
  "dh:revisionSnapshotCSV": "82ba0da2e6564f40261ac2a27ecb657a4ec9b6c9f73899da6c3276ff09ecc041",
  "dcterms:description": "Initial upload",
  "dcterms:license": "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/",
  "dh:publicationDate": "September 30, 2019 08:23",
  "@id": "English-Indices-of-Deprivation/release/2019/revision/1",
  "@type": "dh:Revision",
  "dcat:distribution": {
    "text/csv": {
      "@id": "English-Indices-of-Deprivation/release/2019/revision/1.csv",
      "dcat:mediaType": "text/csv",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019/revision/1.csv"
    },
    "application/json": {
      "@id": "English-Indices-of-Deprivation/release/2019/revision/1.json",
      "dcat:mediaType": "application/json",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019/revision/1.json"
    },
    "application/csvm+json": {
      "@id": "English-Indices-of-Deprivation/release/2019/revision/1-metadata.json",
      "dcat:mediaType": "application/csvm+json",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019/revision/1-metadata.json"
    }
  },
  "dcterms:title": "English Indices of Deprivation 2019 - LSOA Level",
  "@context": [
    "https://cdn.jsdelivr.net/gh/Swirrl/datahost-prototypes@1282114/datahost-ld-openapi/resources/jsonld-context.json",
    {
      "@base": "https://example.org/data/"
    }
  ]
}
</pre>

**Comment by xdrcft8000** [comment]
JSON response for release: 

<pre>

{
  "dcterms:modified": "2024-02-13T09:25:43.096036Z",
  "dcterms:description": "The Indices of Deprivation 2019 provide a set of relative measures of deprivation for small areas (Lower-layer Super Output Areas*) across England, based on seven domains of deprivation. The domains were combined using the following weights to produce the overall Index of Multiple Deprivation (IMD): • Income Deprivation (22.5%) • Employment Deprivation (22.5%) • Education, Skills and Training Deprivation (13.5%) • Health Deprivation and Disability (13.5%) • Crime (9.3%) • Barriers to Housing and Services (9.3%) • Living Environment Deprivation (9.3%) In addition to the Index of Multiple Deprivation and the seven domain indices, there are two supplementary indices: the Income Deprivation Affecting Children Index (IDACI) and the Income Deprivation Affecting Older People Index (IDAOPI). *LSOAs (Lower-layer Super Output Areas) are small areas designed to be of a similar population size, with an average of approximately 1,500 residents or 650 households. There are 32,844 Lower-layer Super Output Areas (LSOAs) in England. They were produced by the Office for National Statistics for the reporting of small area statistics. Following the 2011 Census, the geography of Lower-layer Super Output Areas was revised and the number of areas has increased from 32,482 (as used for the Indices of Deprivation 2010, 2007 and 2004) to 32,844 (used for the Indices of Deprivation 2019). The boundaries of the vast majority (96 per cent) of these 32,844 areas are unchanged since the 2010, 2007 and 2004 Indices. A range of summary measures are available for higher-level geographies including local authority districts and upper-tier local authorities, local enterprise partnerships, and clinical commissioning groups. The Index of Multiple Deprivation, domain indices and the supplementary indices, together with the higher-level geography summaries, are collectively referred to as the Indices of Deprivation. Chapter 3 of the Research Report gives advice on using and interpreting the data. Most of the data used in the Indices of Deprivation 2015 relates to the tax year 2012/13 - see Chapter 4 of the Technical Report for more details. All of the data files and supporting documents for the English Indices of Deprivation 2015 are available at http://www.gov.uk/government/statistics/english-indices-of-deprivation-2019 . For statistical enquiries, please contact: mailto:indices.deprivation@communities.gsi.gov.uk",
  "dcterms:issued": "2024-02-13T09:25:43.096036Z",
  "dcat:inSeries": "https://example.org/data/English-Indices-of-Deprivation",
  "dcterms:license": "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/",
  "dh:hasSchema": "https://example.org/data/English-Indices-of-Deprivation/release/2019/schema",
  "@id": "English-Indices-of-Deprivation/release/2019",
  "dh:hasRevision": [
    "https://example.org/data/English-Indices-of-Deprivation/release/2019/revision/1"
  ],
  "dh:reasonForChange": "Changes between Indices may mean that care should be taken when comparing iterations over time.",
  "@type": "dh:Release",
  "dcat:distribution": {
    "text/csv": {
      "@id": "English-Indices-of-Deprivation/release/2019.csv",
      "dcat:mediaType": "text/csv",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019.csv"
    },
    "application/json": {
      "@id": "English-Indices-of-Deprivation/release/2019.json",
      "dcat:mediaType": "application/json",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019.json"
    },
    "application/csvm+json": {
      "@id": "English-Indices-of-Deprivation/release/2019-metadata.json",
      "dcat:mediaType": "application/csvm+json",
      "dcat:downloadURL": "https://example.org/data/English-Indices-of-Deprivation/release/2019-metadata.json"
    }
  },
  "dcterms:title": "2019",
  "@context": [
    "https://cdn.jsdelivr.net/gh/Swirrl/datahost-prototypes@1282114/datahost-ld-openapi/resources/jsonld-context.json",
    {
      "@base": "https://example.org/data/"
    }
  ]
}

</pre>

**Comment by RickMoynihan** [comment]
This looks good.

I've tried [converting it to RDF](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22dh%3AhasChange%22%3A%5B%22https%3A%2F%2Fexample.org%2Fdata%2FEnglish-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1%2Fcommit%2F1%22%5D%2C%22dh%3AappliesToRelease%22%3A%22https%3A%2F%2Fexample.org%2Fdata%2FEnglish-Indices-of-Deprivation%2Frelease%2F2019%22%2C%22dh%3ArevisionSnapshotCSV%22%3A%2282ba0da2e6564f40261ac2a27ecb657a4ec9b6c9f73899da6c3276ff09ecc041%22%2C%22dcterms%3Adescription%22%3A%22Initial%20upload%22%2C%22dcterms%3Alicense%22%3A%22http%3A%2F%2Fwww.nationalarchives.gov.uk%2Fdoc%2Fopen-government-licence%2Fversion%2F3%2F%22%2C%22dh%3ApublicationDate%22%3A%22September%2030%2C%202019%2008%3A23%22%2C%22%40id%22%3A%22English-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1%22%2C%22%40type%22%3A%22dh%3ARevision%22%2C%22dcat%3Adistribution%22%3A%7B%22text%2Fcsv%22%3A%7B%22%40id%22%3A%22English-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1.csv%22%2C%22dcat%3AmediaType%22%3A%22text%2Fcsv%22%2C%22dcat%3AdownloadURL%22%3A%22https%3A%2F%2Fexample.org%2Fdata%2FEnglish-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1.csv%22%7D%2C%22application%2Fjson%22%3A%7B%22%40id%22%3A%22English-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1.json%22%2C%22dcat%3AmediaType%22%3A%22application%2Fjson%22%2C%22dcat%3AdownloadURL%22%3A%22https%3A%2F%2Fexample.org%2Fdata%2FEnglish-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1.json%22%7D%2C%22application%2Fcsvm%2Bjson%22%3A%7B%22%40id%22%3A%22English-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1-metadata.json%22%2C%22dcat%3AmediaType%22%3A%22application%2Fcsvm%2Bjson%22%2C%22dcat%3AdownloadURL%22%3A%22https%3A%2F%2Fexample.org%2Fdata%2FEnglish-Indices-of-Deprivation%2Frelease%2F2019%2Frevision%2F1-metadata.json%22%7D%7D%2C%22dcterms%3Atitle%22%3A%22English%20Indices%20of%20Deprivation%202019%20-%20LSOA%20Level%22%2C%22%40context%22%3A%5B%22https%3A%2F%2Fcdn.jsdelivr.net%2Fgh%2FSwirrl%2Fdatahost-prototypes%401282114%2Fdatahost-ld-openapi%2Fresources%2Fjsonld-context.json%22%2C%7B%22%40base%22%3A%22https%3A%2F%2Fexample.org%2Fdata%2F%22%7D%5D%7D)

... TODO

**Pull request merged**